src/figuresClasses.ts

```diff
@@ -1,19 +1,86 @@
+type Color = 'red' | 'green' | 'blue';
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle'
+}
+
 export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea: () => number;
+}
 
+function findInvalidValue(...args: number[]): boolean {
+  return args.some((value) => value <= 0);
 }
 
-export class Triangle {
+export class Rectangle implements Figure {
+  public shape: Shape = Shape.Rectangle;
 
+  constructor(
+    public color: Color,
+    public height: number,
+    public width: number,
+  ) {
+    if (findInvalidValue(height, width)) {
+      throw new Error('Width or/and length less than zero');
+    }
+  }
+
+  getArea(): number {
```

Comment:
redundant type casting
```suggestion
  getArea() {
```